packages/pwa-kit-react-sdk/src/ssr/universal/contexts/index.js

```diff
@@ -7,7 +7,43 @@
 
 import React from 'react'
 import DeviceContext from '../device-context'
+import {v4 as uuidv4} from 'uuid'
 
 const ExpressContext = React.createContext()
 
-export {DeviceContext, ExpressContext}
+export {CorrelationIdContext}
+
+const CorrelationIdContext = React.createContext()
+
+const CorrelationIdProvider = ({children, req}) => {
+    // uuid v4 format
+    let correlationId
+    if (req) {
+        // if req.headers['x-amzn-requestid'] is defined, use it
+        // otherwise, generate a new one
+        if (req.headers['x-amzn-requestid']) {
```

Comment:
This header is runtime specific, it should be the `pwa-kit-runtime`'s job to read this header. The `pwa-kit-react-sdk` should have zero knowledge of the underlying runtime/infrastructure.

(BTW, curious, is this a CloudFront thing or API Gateway thing?)

Reply:
Good call, I've moved the logic to `pwa-kit-runtime` to read the amz-* headers.